### PR TITLE
Introduce TickSource abstraction and refactor main loop architecture

### PR DESCRIPTION
**Pull request by lion7**
## Summary

  This PR introduces a TickSource abstraction and refactors the main loop so engine updates are no longer tied directly to a mounted viewport.

  Before this change, the engine loop was effectively driven by Compose frame callbacks from InternalViewport, which meant the engine only advanced when a viewport existed and was actively rendering. That made it difficult or impossible to run Kubriko in headless scenarios such as tests, tools, simulations, servers, editors, or replay systems.

  With this PR, ticking is now provided by a dedicated TickSource, while the viewport is reduced to one possible source of ticks rather than the owner of the game loop.

  ## Why this is needed

  Running the engine headless requires the core update loop to exist independently from UI rendering.

  The previous setup had a few limitations:

  - No viewport meant no frame callbacks, so no engine updates.
  - Engine progression depended on Compose lifecycle/rendering behavior.
  - It was hard to support deterministic/manual ticking for tests or tooling.
  - Fixed-rate or custom non-viewport execution models were not cleanly supported.

  This change separates engine progression from rendering so the same engine instance can run:

  - with a viewport, using viewport frames
  - without a viewport, using a fixed or custom tick source
  - under manual control for deterministic testing and tooling

  ## What changed

  - Added a TickSource abstraction to represent the engine clock.
  - Moved ticking responsibility out of the viewport and into the engine core.
  - Added built-in tick source variants for:
      - viewport-frame driven ticking
      - fixed-rate ticking
      - fixed-frequency ticking
      - manual ticking
  - Updated Kubriko / KubrikoImpl so initialization wires up the selected tick source.
  - Kept InternalViewport as a bridge for viewport-frame ticking instead of the owner of the main loop.

  ## Behavioral impact

  - Engines can now run without attaching a viewport.
  - Headless execution is now a supported architecture instead of an accidental edge case.
  - Viewport-backed execution still works, but rendering is no longer a hard dependency for simulation updates.
  - This also provides a cleaner foundation for tests, editors, server-side simulation, and replay/debug tooling.

  ## Design intent

  The key goal here is separation of concerns:

  - TickSource decides when the engine advances.
  - KubrikoImpl owns update dispatch.
  - InternalViewport only feeds viewport frame ticks when that mode is used.

  That makes the engine more flexible and easier to reason about, while preserving the existing viewport-driven workflow as the default option.

### CODE REVIEW

File: engine/src/commonMain/kotlin/com/pandulapeter/kubriko/Kubriko.kt [hunk 1]
```diff
@@ -44,6 +45,13 @@ sealed interface Kubriko {
      */
     fun <T : Manager> get(managerType: KClass<T>): T
 
+    /**
+     * Initializes this [Kubriko] instance and starts its [TickSource].
+     *
+     * Calling this function multiple times is safe.
+     */
+    fun initialize()
```
Comment (pandulapeter) [inline engine/src/commonMain/kotlin/com/pandulapeter/kubriko/Kubriko.kt]: This might be the only change that I don't instantly agree with.
As most people will rely on the default `ViewportTickSource` that auto-initializes Kubriko the first time the viewport becomes visible, this addition to the public API might be confusing.
Can you maybe find a way to provide an initialization function for the other TickSource types, without making this function public?

**Review by pandulapeter — CHANGES_REQUESTED** [review]
Great contribution, thanks a lot!
I only left one question / comment: let me know if you see a better way to provide an initialization function for the more advanced use cases without changing the API for the simple ones.

File: engine/src/commonMain/kotlin/com/pandulapeter/kubriko/Kubriko.kt (hunk 1 above)
Comment (lion7) [reply engine/src/commonMain/kotlin/com/pandulapeter/kubriko/Kubriko.kt]: I do agree, I was struggling with the same.. I tried a few variations and they all resulted in quite big API changes. This was my attempt with the least impact.

An idea that I still have, is to have a start() method on the tick source which then initializes kubriko. So basically the other way around. For the default viewport based implementation that would result in the same behaviour, in that case it would start the tick source instead of directly initializing kubriko. That means the initialize method on kubriko can stay internal. People that override the tick source can then start the engine explicitly by starting the tick source.

Comment (pandulapeter) [reply engine/src/commonMain/kotlin/com/pandulapeter/kubriko/Kubriko.kt]: That sounds great! Please give it a try. If it doesn't work, let me know and we can merge it like it is.

Comment (lion7) [reply engine/src/commonMain/kotlin/com/pandulapeter/kubriko/Kubriko.kt]: I just pushed that change and it works great. Please re-review, I'm curious what you think.

**Review by pandulapeter — APPROVED** [review]
Great contribution, thank you!
Special thanks for documenting the changes.
